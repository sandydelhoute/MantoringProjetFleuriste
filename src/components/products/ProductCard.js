import React from "react"
import { Link } from "react-router-dom"

import '../../styles/App.css'

const ProductCard = ({product, overitedWitdh}) =>{
    return (
        <Link className="m-2 card" to={`/Produit/${product.id}`} style={{width: `${overitedWitdh}rem`}}>
                <img className="card-img-top" src={product.imageUrl} alt={"Photo du "+ product.name} />
                <div className="card-body">
                    <h5 className="card-title">{product.name}</h5>
                    <p className="card-text">{product.description}</p>
                </div>
        </Link>
    )
}

export default ProductCard