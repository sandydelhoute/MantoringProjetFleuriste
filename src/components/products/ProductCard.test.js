import renderer from 'react-test-renderer';
import ProductCard from './ProductCard';
import {BrowserRouter} from 'react-router-dom';
jest.mock('../../styles/App.css', () => '')
describe('Test render ProductCard', () => {
    it('Test render ProductCard', () => {
        const product = {
            id: 1,
            name: "Produit 1",
            description : 'Produit 1 ...',
            prix : 30.33,
            imageUrl: "test",
            cdate : ""
        }
    const component = renderer.create(<BrowserRouter>  
        <ProductCard product={product}/></BrowserRouter>  ,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    })
});