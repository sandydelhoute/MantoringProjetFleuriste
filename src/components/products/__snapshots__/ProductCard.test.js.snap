// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render ProductCard Test render ProductCard 1`] = `
<a
  className="m-2 card"
  href="/Produit/1"
  onClick={[Function]}
  style={
    {
      "width": "undefinedrem",
    }
  }
>
  <img
    alt="Photo du Produit 1"
    className="card-img-top"
    src="test"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Produit 1
    </h5>
    <p
      className="card-text"
    >
      Produit 1 ...
    </p>
  </div>
</a>
`;
